test/unit/workDiv/src/WorkDivForKernelTest.cpp

```diff
@@ -0,0 +1,225 @@
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+    auto const platform = alpaka::Platform<Acc>{};
+    auto const dev = alpaka::getDevByIdx(platform, 0);
+
+    TestKernelWithManyRegisters kernel;
+    auto const& bundeledKernel = alpaka::KernelBundle(kernel, 200ul);
+
+    // Get hard limits for test
+    auto const& props = alpaka::getAccDevProps<Acc>(dev);
+    Idx const threadsPerGridTestValue = props.m_blockThreadCountMax * props.m_gridBlockCountMax;
+
+    // Test getValidWorkDivForKernel for threadsPerGridTestValue threads per grid
+    auto const workDiv
+        = alpaka::getValidWorkDivForKernel<Acc>(dev, bundeledKernel, Vec{threadsPerGridTestValue}, Vec{1});
+    // Test validity
+    auto const isValid = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, workDiv);
+    CHECK(isValid == true);
+
+    if constexpr(alpaka::accMatchesTags<Acc, alpaka::TagGpuCudaRt>)
+    {
+        // Get calculated threads per block from the workDiv found by examining kernel function
+        auto const threadsPerBlock = workDiv.m_blockThreadExtent.prod();
+        // Get hard limits
+        auto const threadsPerBlockLimit = props.m_blockThreadCountMax;
+        // Depending on the GPU type or the compiler the test below might fail because threadsPerBlock can be equal to
+        // threadsPerBlockLimit, which is the max device limit.
+        CHECK(threadsPerBlock < static_cast<Idx>(threadsPerBlockLimit));
+    }
+    else if constexpr(alpaka::accMatchesTags<Acc, alpaka::TagGpuHipRt>)
+    {
+        // Get calculated threads per block from the workDiv that was found by examining kernel function
+        auto const threadsPerBlock = workDiv.m_blockThreadExtent.prod();
+        // Get hard limits
+        auto const threadsPerBlockLimit = props.m_blockThreadCountMax;
+        // Depending on the GPU type or the compiler the test below might fail because threadsPerBlock can be equal to
+        // threadsPerBlockLimit, which is the max device limit.
+        CHECK(threadsPerBlock == static_cast<Idx>(threadsPerBlockLimit));
+    }
+    else if constexpr(alpaka::accMatchesTags<
+                          Acc,
+                          alpaka::TagCpuSerial,
+                          alpaka::TagCpuThreads,
+                          alpaka::TagCpuOmp2Threads,
+                          alpaka::TagCpuTbbBlocks>)
+    {
+        // CPU must have only 1 thread per block. In other words, number of blocks is equal to number of threads.
+        CHECK(workDiv == WorkDiv{Vec{threadsPerGridTestValue}, Vec{1}, Vec{1}});
+        // Test a new 1D workdiv. Threads per block can not be larger than 1 for CPU. Hence 2 is not valid.
+        auto const& workDiv1DUsingInitList = WorkDiv{Vec{threadsPerGridTestValue / 2}, Vec{2}, Vec{1}};
+        auto const isWorkDivValidForCpu
+            = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, workDiv1DUsingInitList);
+        CHECK(isWorkDivValidForCpu == false);
+    }
```

Comment:
please add a else branch with an exception, to see if we miss an accelerator.

Reply:
added, thanks.